nupic/research/connections.py

```diff
@@ -23,14 +23,28 @@
 
 
 
+class SynapseData(object):
+
+
+  __slots__ = ("segment", "presynapticCell", "permanence")
+
+
+  def __init__(self, segment, presynapticCell, permanence):
+    self.segment = segment
+    self.presynapticCell = presynapticCell
+    self.permanence = permanence
+
+
+  def __eq__(self, other):
+    return (self.segment, self.presynapticCell, self.permanence) == other
+
+
+
 class Connections(object):
   """
   Class to hold data representing the connectivity of a collection of cells.
   """
 
-  SynapseData = namedtuple("SynapseData", ["segment",
-                                           "presynapticCell",
-                                           "permanence"])
 
```

Comment:
I agree that the fix needs to be in NuPIC Studio. I'm not sure about how easy it will be to fix, though. As internal architectures change in NuPIC, this could continue to be a problem. It really won't be solved until we get to NuPIC 1.0 and an official public API.